app/questionnaire/navigation.py

```diff
@@ -40,14 +40,25 @@ def build_navigation(self, current_group_id, current_group_instance):
             logger.debug("Building frontend navigation for group %s", group['id'])
 
             repeating_rule = SchemaHelper.get_repeat_rule(group)
+            skip_group = self._should_skip_group(current_group_instance, group)
+
+            if not skip_group:
+                if repeating_rule:
+                    navigation += self._build_repeating_navigation(repeating_rule, group, current_group_id,
```

Comment:
Ah, okay sorry I hadn't spotted that. In which case `extend` will work. I just think making it explicit is a bit clearer than using the operator. Not a big deal, up to you if you want to change it.